### PR TITLE
Include diagonal control

### PR DESCRIPTION
**Pull request by cua-cua**

### CODE REVIEW

File: control.c [hunk 1]
```diff
@@ -16,8 +16,10 @@ static volatile bool collision_detected_signal;
 static volatile bool motor_control_enabled_signal;
 static volatile bool side_sensors_control_enabled;
 static volatile bool front_sensors_control_enabled;
+static volatile bool diagonal_enabled;
 static volatile float side_sensors_integral;
 static volatile float front_sensors_integral;
+static volatile float diagonal_integral;
```
Comment (Peque) [inline control.c]: I would use a more consistent naming, like `diagonal_sensors_control_enabled` and `diagonal_sesors_integral`.

File: solve.c [hunk 2]
```diff
@@ -172,8 +172,9 @@ static void execute_movement_sequence(char *sequence, float force,
 		case MOVE_RIGHT_DIAGONAL:
 			distance += get_move_turn_before(movement);
 			side_sensors_control(false);
-			parametric_move_front(
-			    distance,
+			LOG_ERROR("solve diagonal");
```
Comment (Peque) [inline solve.c]: Remember to remove this log message!

File: walls.c [hunk 3]
```diff
@@ -103,6 +104,27 @@ float get_front_sensors_error(void)
 	return distance[SENSOR_FRONT_LEFT_ID] - distance[SENSOR_FRONT_RIGHT_ID];
 }
 
+/**
+ * @brief Calculate and return the diagonal sensors error.
+ *
+ * Taking into account that robot is running between diagonal pillars, this
+ * function returns the distance is moved from the center of the diagonal.
```
Comment (Peque) [inline walls.c]: This description is not correct. It just tells whether or not we are too close to a pillar.

**Review by Peque — CHANGES_REQUESTED** [review]

**Comment by Peque** [comment]
@cua-cua Maybe we could take the chance and close https://github.com/Bulebots/bulebule/issues/246 too?

With the same strategy: enable "diagonals" corrections just to avoid crashing into a pillar.

File: control.c (hunk 1 above)
Comment (cua-cua) [reply control.c]: 90e89df9ff933

File: solve.c (hunk 2 above)
Comment (cua-cua) [reply solve.c]: 90e89df9ff933

**Comment by cua-cua** [comment]
> @cua-cua Maybe we could take the chance and close [Bulebots/bulebule#246](https://github.com/Bulebots/bulebule/issues/246) too?
> 
> With the same strategy: enable "diagonals" corrections just to avoid crashing into a pillar.

I will create a new pull request ;)

File: walls.c (hunk 3 above)
Comment (cua-cua) [reply walls.c]: 4366a657

Comment (Peque) [reply walls.c]: > This function returns an error if the robot is too close to a pillar. The error is calculated as the difference between the sensed distance and a minimum distance threshold.

Comment (cua-cua) [reply walls.c]: acf79a2f

**Comment by cua-cua** [comment]
New pull request #17 with rebase against master

**Pull request closed without merging**